test/dynamo/test_misc.py

```diff
@@ -11359,6 +11359,117 @@ def get_rng():
         self.assertEqual(expected, actual)
         self.assertGreater(po.call_count, 0)
 
+    def test_graph_output_count(self):
+        test_case_self = self
+
+        class AssertNumOutputBackend:
+            """
+            A backend that checks the number of output for compiled graph, and
+            return the graph as is.
+            """
+
+            def __init__(self, expected_num_output: int):
+                self.expected_num_output = expected_num_output
+
+            def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+                outputs = gm(*example_inputs)
+                test_case_self.assertEqual(self.expected_num_output, len(outputs))
+                return gm
+
+        @torch.compile(backend=AssertNumOutputBackend(2))
+        def test_returning_nested_func_with_captured_tensor():
```

Comment:
These should probably be separate tests?

Reply:
Lemme try that, I think I was trying to avoid creating too many functions or something.